nuttx/arch/arm/src/stm32/Kconfig

```diff
@@ -1690,7 +1689,184 @@ config STM32_ADC_INIT
 		Create a /dev/adcX for each ADC device.
 
 if STM32_ADC_INIT
-if STM32_ADC1
+if STM32_ADC1 && STM32_STM32L4
+menu "ADC1 Device Initialization"
+
+config STM32_ADC1_CHAN0_INIT
+	bool "ADC1 Channel 0 (IN1-PC0)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC || ARCH_CHIP_STM32L431CB || ARCH_CHIP_STM32L431CC)
+config STM32_ADC1_CHAN1_INIT
+	bool "ADC1 Channel 1 (IN2-PC1)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC || ARCH_CHIP_STM32L431CB || ARCH_CHIP_STM32L431CC)
+config STM32_ADC1_CHAN2_INIT
+	bool "ADC1 Channel 2 (IN3-PC2)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC || ARCH_CHIP_STM32L431CB || ARCH_CHIP_STM32L431CC)
+config STM32_ADC1_CHAN3_INIT
+	bool "ADC1 Channel 3 (IN4-PC3)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC || ARCH_CHIP_STM32L431CB || ARCH_CHIP_STM32L431CC)
+config STM32_ADC1_CHAN4_INIT
+	bool "ADC1 Channel 4 (IN5-PA0)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC)
+config STM32_ADC1_CHAN5_INIT
+	bool "ADC1 Channel 5 (IN6-PA1)"
+	default n
+config STM32_ADC1_CHAN6_INIT
+	bool "ADC1 Channel 6 (IN7-PA2)"
+	default n
+config STM32_ADC1_CHAN7_INIT
+	bool "ADC1 Channel 7 (IN8-PA3)"
+	default n
+config STM32_ADC1_CHAN8_INIT
+	bool "ADC1 Channel 8 (IN9-PA4)"
+	default n
+config STM32_ADC1_CHAN9_INIT
+	bool "ADC1 Channel 9 (IN10-PA5)"
+	default n
+config STM32_ADC1_CHAN10_INIT
+	bool "ADC1 Channel 10 (IN11-PA6)"
+	default n
+	depends on !(ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+config STM32_ADC1_CHAN11_INIT
+	bool "ADC1 Channel 11 (IN12-PA7)"
+	default n
+	depends on !(ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+config STM32_ADC1_CHAN12_INIT
+	bool "ADC1 Channel 12 (IN13-PC4)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC || ARCH_CHIP_STM32L431CB || ARCH_CHIP_STM32L431CC \
+	|| ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+config STM32_ADC1_CHAN13_INIT
+	bool "ADC1 Channel 13 (IN14-PC5)"
+	default n
+	depends on !(ARCH_CHIP_STM32L431KB || ARCH_CHIP_STM32L431KC || ARCH_CHIP_STM32L431CB || ARCH_CHIP_STM32L431CC \
+	|| ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+config STM32_ADC1_CHAN14_INIT
+	bool "ADC1 Channel 14 (IN15-PB0)"
+	default n
+	depends on !(ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+config STM32_ADC1_CHAN15_INIT
+	bool "ADC1 Channel 15 (IN16-PB1)"
+	default n
+	depends on !(ARCH_CHIP_STM32L433CB || ARCH_CHIP_STM32L433CC)
+endmenu
+endif
+endif
```

Comment:
I can't find a Kconfig example were ELSE is used, just multiple groupings of IF to ENDIF